src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -365,9 +365,8 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
   // this is of crucial importance when populating the table in parallel
   protected final def index(hcode: Int) = {
     val ones = table.length - 1
-    val improved = improve(hcode, seedvalue)
-    val shifted = (improved >> (32 - java.lang.Integer.bitCount(ones))) & ones
-    shifted
+    val exponent = Integer.numberOfLeadingZeros(ones)
```

Comment:
since `table.length` is a power of two, `table.length - 1` is composed of ones. The bitCount of that is the number of ones (= the exponent of the original power of 2), subtracted from 32 is the number of trailing zeroes.

The benchmark indicates that this is smaller and faster:
```java
public static class Test {
    final int[] squares = {1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16384, 32768, 65536, 131072, 262144, 524288, 1048576, 2097152, 4194304, 8388608, 16777216, 33554432, 67108864, 134217728, 268435456, 536870912, 1073741824};

    @Benchmark
    public void bitCount(Blackhole bh) {
        for (int i = 0; i < squares.length; i++) {
            int leadingZeros = Integer.SIZE - Integer.bitCount(squares[i] - 1);
            bh.consume(leadingZeros);
        }
    }

    @Benchmark
    public void numberOfLeadingZeros(Blackhole bh) {
        for (int i = 0; i < squares.length; i++) {
            int leadingZeros = Integer.numberOfLeadingZeros(squares[i] - 1);
            bh.consume(leadingZeros);
        }
    }
}
```

```java
Benchmark                        Score         Error  Units
Test.bitCount              8746665.311  ± 206703.549  ops/s
Test.numberOfLeadingZeros  9255031.968  ± 114967.712  ops/s
```
(`ops/s`, greater is better)

Note: this affect all hashing operations!